crates/ruff_linter/src/rules/flake8_markupsafe/rules/unsafe_markup_use.rs

```diff
@@ -0,0 +1,102 @@
+use ruff_python_ast::ExprCall;
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::name::QualifiedName;
+use ruff_text_size::Ranged;
+
+use crate::{checkers::ast::Checker, settings::LinterSettings};
+
+/// ## What it does
+/// Checks for non-literal strings being passed to [`markupsafe.Markup`].
+///
+/// ## Why is this bad?
+/// `markupsafe.Markup` does not perform any escaping, so passing dynamic
+/// content, like f-strings, variables or interpolated strings will potentially
+/// lead to XSS vulnerabilities.
+///
+/// Instead you should interpolate the [`markupsafe.Markup`] object.
```

Comment:
Not sure it's worth mentioning mako support, since mako support was not part of flake8, but rather had to be invoked manually by directly calling the mako module within the plugin. But mentioning the i18n deviation makes sense to me.